typing/ctype.ml

```diff
@@ -1969,10 +1970,11 @@ let occur_univar env ty =
                    in this position. Physical expansion, as done in `occur`,
                    would be costly here, since we need to check inside
                    object and variant types too. *)
-                if not Variance.(eq v null) then occur_rec bound t)
+                if Variance.(if inj_only then mem Inj v else not (eq v null))
+                then occur_rec bound t)
               tl td.type_variance
           with Not_found ->
-            List.iter (occur_rec bound) tl
+            if not inj_only then List.iter (occur_rec bound) tl
```

Comment:
It looks like we assume, when when we don't know the variance of a constructor, that its parameters are not injective. Is this assumption correct?

Reply:
`Inj` means "known to be injective". If the type is unknown, its parameters cannot be known to be injective.